### PR TITLE
fix: handle unknown CSS properties like `@angular/platform-server`

### PR DESCRIPTION
**Pull request by alan-agius4**
Context: https://github.com/crisbeto/angular/blob/a1e294c55cdbe47fab0e620f9898f8ce917c17a0/packages/platform-server/src/server_renderer.ts#L303

This is needed to try to remove the `ServerRendererFactory2`.

### CODE REVIEW

File: lib/CSSStyleDeclaration.js [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
-  var result = { property: Object.create(null), priority: Object.create(null) };
-  parser.addListener("property", function(e) {
-    if (e.invalid) return; // Skip errors
-    result.property[e.property.text] = e.value.text;
-    if (e.important) result.priority[e.property.text] = 'important';
-  });
-  s = (''+s).replace(/^;/, '');
-  parser.parseStyleAttribute(s);
-  return result;
+  const property = {};
```
Comment (AndrewKushnir) [inline lib/CSSStyleDeclaration.js:16]: FYI, I think we can reuse the code of the styling parser from Angular: https://github.com/angular/angular/blob/main/packages/compiler/src/render3/view/style_parser.ts.

I believe the current implementation may miss cases where value has quotes, etc.

See more tricky cases in https://github.com/angular/angular/blob/main/packages/compiler/test/render3/style_parser_spec.ts.

Comment (alan-agius4) [reply lib/CSSStyleDeclaration.js:16]: Will take a look tomorrow a

File: lib/style_parser.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+Object.defineProperty(exports, "__esModule", { value: true });
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply lib/style_parser.js:13]: Moved it to be more visible

[the remark this answers is not in the document]
Comment (AndrewKushnir) [reply lib/style_parser.js:13]: I definitely missed it because if was above the license header :) Thanks for moving it!

**Pull request merged**